scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,304 @@
+    print 'SIGINT signo = %s, frame = %s' % (_signo, _frame)
+    check(['docker', 'stop', CONTAINER])
+
+
+def main(args):
+    """Set up env, trigger runner, handle termination. """
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+
+    # platform envs
+    if args.platform:
+        load_env(test_infra('platforms/' + args.platform + '.env'))
+
+    # job envs
+    if args.env_file:
+        load_env(test_infra('jobs/' + args.env_file + '.env'))
+
+    # job env templates
+    if args.env_template:
+        load_template(test_infra('jobs/' + args.env_template + '.env'), args)
+
+    # Other args Override
+    if args.zone:
+        os.environ['ZONE'] = args.zone
+        os.environ['KUBE_GCE_ZONE'] = args.zone
+    if args.project:
+        os.environ['PROJECT'] = args.project
+    if args.use_cert_auth:
+        os.environ['CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE'] = 'true'
+
+    # Boilerplate envs
+    # Assume we're upping, testing, and downing a cluster
+    os.environ['E2E_UP'] = os.environ.get('E2E_UP') or 'true'
+    os.environ['E2E_TEST'] = os.environ.get('E2E_TEST') or 'true'
+    os.environ['E2E_DOWN'] = os.environ.get('E2E_DOWN') or 'true'
+
+    e2e_name = os.environ.get('E2E_NAME') or 'bootstrap-e2e'
+
+    # Skip gcloud update checking
+    os.environ['CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK'] = 'true'
+    # Use default component update behavior
+    os.environ['CLOUDSDK_EXPERIMENTAL_FAST_COMPONENT_UPDATE'] = 'false'
+
+    # AWS variables
+    os.environ['KUBE_AWS_INSTANCE_PREFIX'] = e2e_name
+
+    # GCE variables
+    os.environ['INSTANCE_PREFIX'] = e2e_name
+    os.environ['KUBE_GCE_NETWORK'] = e2e_name
+    os.environ['KUBE_GCE_INSTANCE_PREFIX'] = e2e_name
+
+    # GKE variables
+    os.environ['CLUSTER_NAME'] = e2e_name
+    os.environ['KUBE_GKE_NETWORK'] = e2e_name
+
+    # Get golang into our PATH so we can run e2e.go
+    os.environ['PATH'] += ':/usr/local/go/bin'
+
+    # dockerized-e2e-runner goodies setup
+    workspace = os.environ.get('WORKSPACE') or os.getcwd()
+    repo = os.environ.get('REPO_DIR') or os.getcwd()
+    artifacts = '%s/_artifacts' % workspace
+    os.makedirs(artifacts)
+
+    # TODO(ixdy): remove when all jobs are setting these vars using Jenkins credentials
+    gce_key = '/var/lib/jenkins/gce_keys/google_compute_engine'
+    if os.path.isfile(gce_key):
+        os.environ['JENKINS_GCE_SSH_PRIVATE_KEY_FILE'] = gce_key
+
+    gce_key_pub = '/var/lib/jenkins/gce_keys/google_compute_engine.pub'
+    if os.path.isfile(gce_key_pub):
+        os.environ['JENKINS_GCE_SSH_PUBLIC_KEY_FILE'] = gce_key_pub
+
+
+    e2e_image_tag = 'v20170104-9031f1d'
+    e2e_image_tag_override = '%s/hack/jenkins/.kubekins_e2e_image_tag' % workspace
+    if os.path.isfile(e2e_image_tag_override):
+        with open(e2e_image_tag_override, 'r') as tag:
+            e2e_image_tag = tag.read()
+
+    docker_env_ignore = [
```

Comment:
Yeah agree. I'll make it a separate PR, I'd rather make a job works as previous first, and dump down what do we really need here.